src/main/java/com/laytonsmith/core/functions/DataHandling.java

```diff
@@ -318,11 +353,181 @@ public Mixed exec(Target t, Environment env, Mixed... args) throws CancelCommand
 			return v;
 		}
 
+		@Override
```

Comment:
The scope graph I've created for this analysis does not follow the control flow in a way you'd need to find the latest assign(s) of a variable. Imagine:
```
int @a = 1;
if(...) {
    @a = 2;
}
use(@a);
```
Here, usage of `@a` can really only resolve to the declaration, since the second assign is out of scope according to the scope graph. Having that said, the scope graph implementation does support creating namespace-specific scope edges, so what we can do is:
```
1. Put each variable assign as a declaration into a new namespace.
2. Add namespace-specific edges to control flow functions for this namespace, such that if/else branches are accessible after the if() call etc.
3. For each variable reference, check whether the variable assign(s) it resolves to is/are static (and have the same value in case there's multiple).
```
It's funny how I thought that this wasn't possible with this analysis, but actually it is.